### ISSUE TITLE
Sudo-edit does not work on remote files

### ISSUE THREAD

**Issue by roblabla** [syl20bnr/spacemacs#11329]
#### Description :octocat:
sudo-edit does not work on remote (TRAMP) files. It crashes with Wrong number of arguments.

#### Reproduction guide :beetle:
- Start Emacs
- Open a remote file (for instance, on ssh)
- Call spacemacs/sudo-edit (SPC f E)

*Observed behaviour:* :eyes: :broken_heart:
Wrong number of arguments (6 . 7), 5

*Expected behaviour:* :heart: :smile:
A new buffer is opened with the file in sudo.

#### System Info :computer:
- OS: darwin
- Emacs: 26.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 9865e7731)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: ivy
- Layers:
```elisp
(yaml rust ivy auto-completion better-defaults emacs-lisp git neotree
      (shell :variables shell-default-shell 'eshell shell-enable-smart-eshell t shell-default-height 30 shell-default-position 'bottom)
      syntax-checking version-control)
```
- System configuration features: JPEG NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS THREADS

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (wrong-number-of-arguments (6 . 7) 5)
  tramp-make-tramp-file-name(#("sshx" 0 4 (escaped t)) #("roblabla" 0 8 (escaped t)) #("roblab.la" 0 9 (escaped t)) nil nil)
  (let* ((new-hop (tramp-make-tramp-file-name parsed-method parsed-user parsed-host nil parsed-hop)) (new-hop (substring new-hop 1 -1)) (new-hop (concat new-hop "|")) (new-fname (tramp-make-tramp-file-name "sudo" "root" parsed-host parsed-localname new-hop))) new-fname)
  (let* ((parsed (tramp-dissect-file-name fname)) (parsed-method (tramp-file-name-method parsed)) (parsed-user (tramp-file-name-user parsed)) (parsed-domain (tramp-file-name-domain parsed)) (parsed-host (tramp-file-name-host parsed)) (parsed-port (tramp-file-name-port parsed)) (parsed-localname (tramp-file-name-localname parsed)) (parsed-hop (tramp-file-name-hop parsed))) (ignore parsed-method parsed-user parsed-domain parsed-host parsed-port parsed-localname parsed-hop) (if (equal parsed-user "root") (progn (error "Already root!"))) (let* ((new-hop (tramp-make-tramp-file-name parsed-method parsed-user parsed-host nil parsed-hop)) (new-hop (substring new-hop 1 -1)) (new-hop (concat new-hop "|")) (new-fname (tramp-make-tramp-file-name "sudo" "root" parsed-host parsed-localname new-hop))) new-fname))
  (with-parsed-tramp-file-name fname parsed (if (equal parsed-user "root") (progn (error "Already root!"))) (let* ((new-hop (tramp-make-tramp-file-name parsed-method parsed-user parsed-host nil parsed-hop)) (new-hop (substring new-hop 1 -1)) (new-hop (concat new-hop "|")) (new-fname (tramp-make-tramp-file-name "sudo" "root" parsed-host parsed-localname new-hop))) new-fname))
  (if (not (tramp-tramp-file-p fname)) (concat "/sudo:root@localhost:" fname) (with-parsed-tramp-file-name fname parsed (if (equal parsed-user "root") (progn (error "Already root!"))) (let* ((new-hop (tramp-make-tramp-file-name parsed-method parsed-user parsed-host nil parsed-hop)) (new-hop (substring new-hop 1 -1)) (new-hop (concat new-hop "|")) (new-fname (tramp-make-tramp-file-name "sudo" "root" parsed-host parsed-localname new-hop))) new-fname)))
  (find-file (if (not (tramp-tramp-file-p fname)) (concat "/sudo:root@localhost:" fname) (with-parsed-tramp-file-name fname parsed (if (equal parsed-user "root") (progn (error "Already root!"))) (let* ((new-hop (tramp-make-tramp-file-name parsed-method parsed-user parsed-host nil parsed-hop)) (new-hop (substring new-hop 1 -1)) (new-hop (concat new-hop "|")) (new-fname (tramp-make-tramp-file-name "sudo" "root" parsed-host parsed-localname new-hop))) new-fname))))
  (let ((fname (if (or arg (not buffer-file-name)) (read-file-name "File: ") buffer-file-name))) (find-file (if (not (tramp-tramp-file-p fname)) (concat "/sudo:root@localhost:" fname) (with-parsed-tramp-file-name fname parsed (if (equal parsed-user "root") (progn (error "Already root!"))) (let* ((new-hop (tramp-make-tramp-file-name parsed-method parsed-user parsed-host nil parsed-hop)) (new-hop (substring new-hop 1 -1)) (new-hop (concat new-hop "|")) (new-fname (tramp-make-tramp-file-name "sudo" "root" parsed-host parsed-localname new-hop))) new-fname)))))
  spacemacs/sudo-edit(nil)
  funcall-interactively(spacemacs/sudo-edit nil)
  call-interactively(spacemacs/sudo-edit nil nil)
  command-execute(spacemacs/sudo-edit)
```

**Comment by Caribpa** [comment]
I've fixed this issue on #12867 

It should work for any tramp method that allows multi-hopping in either `tramp-login-program`, `tramp-copy-program`, or its own tramp method name. Note that my fix will try to get a tramp method in the order mentioned previously.

**Comment by duianto** [comment]
The following PRs (possibly modified) changes are now on the develop branch:
[spacemacs/sudo-edit] Fix sudo-edit on TRAMP & [spacemacs/counsel-search] Fix wrong initial directory #12867
